packages/atomic/src/components/search/generated-answer/atomic-generated-answer-feedback/atomic-generated-answer-feedback-modal.tsx

```diff
@@ -0,0 +1,275 @@
+import {GeneratedAnswer} from '@coveo/headless';
+import {GeneratedAnswerFeedback} from '@coveo/headless/dist/definitions/features/generated-answer/generated-answer-analytics-actions';
+import {
+  Component,
+  State,
+  h,
+  Watch,
+  Prop,
+  Element,
+  Event,
+  EventEmitter,
+} from '@stencil/core/internal';
+import CloseIcon from '../../../../images/close.svg';
+import Success from '../../../../images/success.svg';
+import {
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../../utils/initialization-utils';
+import {updateBreakpoints} from '../../../../utils/replace-breakpoint';
+import {once, randomID} from '../../../../utils/utils';
+import {Button} from '../../../common/button';
+import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
+
+@Component({
+  tag: 'atomic-generated-answer-feedback-modal',
+  styleUrl: 'atomic-generated-answer-feedback-modal.pcss',
+  shadow: true,
+})
+export class AtomicGeneratedAnswerFeedbackModal
+  implements InitializableComponent
+{
+  @InitializeBindings() public bindings!: Bindings;
+  @Element() public host!: HTMLElement;
+
+  @Prop({reflect: true, mutable: true}) isOpen = false;
+  @Prop() generatedAnswer!: GeneratedAnswer;
+
+  @State() public error!: Error;
+  @State() currentAnswer?: GeneratedAnswerFeedback | 'other' | undefined;
+  @State() feedbackSubmitted: boolean = false;
+
+  private readonly formId = randomID(
+    'atomic-generated-answer-feedback-modal-form-'
+  );
+  private detailsInputRef?: HTMLTextAreaElement;
+
+  @Event() feedbackSent!: EventEmitter;
+
+  @Watch('isOpen')
+  watchToggleOpen(isOpen: boolean) {
+    if (isOpen) {
+      this.generatedAnswer.openFeedbackModal();
+    }
+  }
+
+  private setIsOpen(isOpen: boolean) {
+    this.isOpen = isOpen;
+  }
+
+  private close() {
+    this.feedbackSubmitted = false;
+    this.setIsOpen(false);
+    this.generatedAnswer.closeFeedbackModal();
+  }
```

Comment:
I guess closing the modal should clear `currentAnswer` and the details input box.